aiohttp/resolver.py

```diff
@@ -26,6 +27,15 @@ def __init__(self, loop=None):
         self._loop = loop
 
     async def resolve(self, host, port=0, family=socket.AF_INET):
+        if family == socket.AF_UNSPEC:
+            hosts = await self._resolve(host, port, family=socket.AF_INET)
+            hosts.extend(await self._resolve(
+                host, port, family=socket.AF_INET6))
```

Comment:
It looks like `AF_UNSPEC` would already return all IPv6 and IPv4 addresses: http://man7.org/linux/man-pages/man3/getaddrinfo.3.html

So what is the point of separate calls for those?

Reply:
Yes, you are right. Seems this hack should be applied only in those resolve methods who don't support resolving both families using one API request.

Reply:
Oh.. I see: stdlib call is synchronous, but running in the executor https://github.com/python/cpython/blob/bec2372b7e1da5dfdbadaf242aa8e994b164cace/Lib/asyncio/base_events.py#L771

Reply:
Seems like cares doesn't support AF_UNSPEC

Reply:
@asvetlov what will happen if the server is bound to `socket.AF_UNIX`, but the client code will call this function with `socket.AF_UNSPEC`, for example (or smth else mismatching)?

Reply:
@asvetlov "cares doesn't support AF_UNSPEC "?

I'm not sure I've got what you mean here. Could you please add more detail?

Reply:
The resolver is not used for unix sockets.
async resolver doesn't support AF_UNSPEC, it requires two explicit calls for DNS resolving -- one for IPv4 and another one for IPv6.
At least I've tried to do it in the single call but failed

Reply:
@asvetlov according to https://github.com/python/cpython/blob/bec2372b7e1da5dfdbadaf242aa8e994b164cace/Lib/asyncio/base_events.py#L771 it just bypasses all args to underlying synchronous `socket.getaddrinfo`, which supports `UNSPEC` https://github.com/python/cpython/blob/master/Modules/socketmodule.c#L6189 and bypasses that into underlying syscall.

The only thing I can think of if a small possibility of behaving differently in non-UNIX envs, but anyway I don't understand how could it fail for you. Can you please post a traceback here?